src/services/bundle-descriptor-converter-service.ts

```diff
@@ -65,6 +81,45 @@ export class BundleDescriptorConverterService {
     this.generateBundleYamlDescriptor(bundleDescriptor)
   }
 
+  public processThumbnail(): BundleThumbnailInfo {
```

Comment:
I suggest to extract thumbnail logic in a separate service

Reply:
logic now moved to new service `bundle-thumbnail-service.ts`